fs/base.py

```diff
@@ -1652,7 +1652,7 @@ def hash(self, path, name):
             fs.errors.UnsupportedHash: If the requested hash is not supported.
 
         """
-        _path = self.validatepath(path)
+        self.validatepath(path)
```

Comment:
@willmcgugan does `_path` being an unused variable hint that the `openbin` function below should have been using `_path` instead of `path` ?

Reply:
flake8 flag it with red underline.

<img width="415" alt="Screenshot 2019-08-08 at 22 42 31" src="https://user-images.githubusercontent.com/4280312/62739830-d0bf3f80-ba2d-11e9-9be3-8e876d561401.png">

Reply:
bad code never escape

Reply:
@chfw Yeah, but I was wondering if it means that the code _should_ have been doing `self.openbin(_path)` ? (i.e. perhaps it's only an unused variable because of a typo?)

Reply:
OK. got you. Let's wait for @willmcgugan

Reply:
> @willmcgugan does _path being an unused variable hint that the openbin function below should have been using _path instead of path ?

That's generally the case, but in this instance the return value isn't needed because `openbin` will also call `validatepath`. The right call here would be to discard the return value but keep the call.

If you're wondering why call `validatepath` at all if its called in `openbin`, its so that path errors take priority over the unsupported hash error...